### PR TITLE
Export functions and minor style fixes

### PR DESCRIPTION
**Pull request by ninedraft**
Thanks for the very useful package!🙌

I'm suggesting slight changes associated with the cycles, perhaps purely stylistic. I also exported functions so that they can be used outside the package!

### CODE REVIEW

**Comment by anna-oake** [comment]
Hey, this is just a test snippet, don't take it so seriously.
And actually, if you really want to use this code (as I do in my Go library), you wouldn't need these functions exported, because your library uses them internally.

Thanks for style fixes, I can merge them if you unexport functions back.

File: cipher.go [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
 
-func cipherInit(key []byte) []byte {
-	perm := make([]uint8, 256)
-	var ia int
+func CipherInit(key []byte) []byte {
+	var perm = make([]uint8, 0, 256)
+	perm = append(perm, []uint8(key)...)
 	var j uint8
 	keyLen := uint8(len(key))
-	for i := 0; i <= 255; i++ {
```
Comment (anna-oake) [inline cipher.go:8]: This is not a conversion. We are filling an array with 0..255 values here.

Comment (ninedraft) [reply cipher.go:8]: Ooops
My bad!

**Comment by ninedraft** [comment]
> you wouldn't need these functions exported, because your library uses them internally.

Could you clarify that point? You offer me not to import your package, but to copy the code from it?

**Comment by anna-oake** [comment]
> You offer me not to import your package, but to copy the code from it?

@ninedraft, yes, exactly. This repo is not a Go package. This repo is just a protocol description and cipher.go file is a snippet copied from my library with intention to show how cipher functions work in Xiaomi BLE.

Maybe I'll release my library later so you'll just use it like k := kettle.Connect("mac"), k.SetTemperature(), etc.

But now this is just a simple snippet.

**Comment by ninedraft** [comment]
Ok :(

**Comment by anna-oake** [comment]
> Ok :(

Hey I'm fine with your style fixes, etc.
Let's just keep functions private.

**Comment by ninedraft** [comment]
> Let's just keep functions private.
Done

**Comment by aprosvetova** [comment]
Thanks!

Now I see that your code returns different values :D 

You can try these functions from original code and your code, then compare outputs, they'll differ.
Not sure what exactly breaks this.

**Comment by aprosvetova** [comment]
Outputs from `cipherInit` are same.

**Comment by anna-oake** [comment]
Oh and I forgot that we don't need characteristics.go.
All characteristics are listed in protocol description. As I explained, this repo is just a description, no code here. Furthermore, currantlabs/ble will not work with characteristics as numbers.

**Pull request merged**